### DISCUSSION TITLE
Calculating mod multipliers based on community survey (n = 68)

### DISCUSSION THREAD

**Discussion by MaklovitzLazer** [ppy/osu#26999] [Ideas]
### Abstract / tldr:
![image](https://github.com/ppy/osu/assets/158756313/9cde2aea-a919-4375-bbb5-419b02ad5c1f)

### Methodology and results:
Based on the raw data from "What % accuracy should X scores tie NM SS scores?" survey assessed in osu! community (https://docs.google.com/forms/d/1ccBSyGq9tN_phmJIFetGOxdfBlHtB2RHNtyMLFJuR1A) I calculated mod multipliers for HD, HR, DT and FL in a way that they match the mean expectation of 68 voters. I got surveys' raw data from Elijah (**sevenend7** on discord).
1) I calculated mean results of What accuracy should X scores tie NM SS scores?
![image](https://github.com/ppy/osu/assets/158756313/a16d9677-47b6-4eee-be31-8e1989a4594a)
2) Fitted bell curves (assumed normal distribution) to that 
![image](https://github.com/ppy/osu/assets/158756313/87e840c3-4390-457c-b9fc-e9c3f540ecb4)
3) Used lazer score formula from ppy github https://github.com/ppy/osu/blob/ef2e2302d4cf3f049c5637527f99b311d999c43b/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L380
![image](https://github.com/ppy/osu/assets/158756313/d8c8e394-e308-4d1f-8e69-7e6d557ad213)
to evaluate mod multipliers, so that for the cutoff accuracy (from the survey) of the FC score with given mod gives as much score as NM SS, which is 1 000 000 (assumed bonus portion is 0 & assumed FC with no sliderends dropped). The resulted formula is
$m = \frac{2}{a + a^5}$
a - mean accuracy from the survey
m - corresponding mod multiplier
4) Calculated mod multipliers for HD, HR, DT and FL and for the most popular mod combinations.
![image](https://github.com/ppy/osu/assets/158756313/0c80541b-ba8f-476d-bcf2-9b75fbbf623a)
Here the exactly fitted values, if someone wants to use them for their own research:
HD | 1.0815889623755
HR | 1.14705534707846
FL | 1.22892198047254
DT | 1.26891860957178
5) Used score formula to calculate FC score in function of accuracy for the most popular mod multipliers
![image](https://github.com/ppy/osu/assets/158756313/a7cfcbcd-c1e4-425f-a210-99d7b250b4fc)

### Conclusions:

Due to the new accuracy scaling
stable score ~ acc (if FC)
lazer score ~ $\frac{acc^5 + acc}{2}$ (if FC)
people find ALL mod multipliers underrated, which causes weird looking leaderboards (for example HDDT SS worth as much score as 98.0% HDHRDT FC) and a lot of frustration in leaderboard playing community. My take on this topic can be found here:
https://twitter.com/Maklovitz_osu/status/1752436258465259745
This analysis showed that we can easily evaluate the best mod multipliers, based on what players think about comparing an FC with a mod to a nomod SS. The upgraded version of this experiment would require asking thousands of active osu! community members (maybe even vote weight based on voter's pp rank & ranked score/# of lb scores), which would be hard without an official survey from peppy. High diversity of answers and independence of the voters is the key factor to benefit from the intelligence of the crowd. 

Thanks to Elijah (SevenEnd7) for inspring me to do this analysis. His tweet that started all this:
https://twitter.com/SevenEnd7/status/1753668111998546200

WitherFlower's spreadsheet that works alomst the same as my calculations:
You can make a copy to play with different values: https://docs.google.com/spreadsheets/d/1iv7ptvppa9n-cBWrUlSDIlKZgjM0kf9LYMAMPpnSfow/edit?usp=sharing

**More discussion about this topic can be found under this tweet:**
https://twitter.com/Maklovitz_osu/status/1753743401990697415
Note that the results included in my tweet are slightly incorrect, which I explain in the comment:
https://twitter.com/Maklovitz_osu/status/1753755502880715089

**Comment by MaklovitzLazer** [comment]
Additional thoughts: In the survey we shouldn't we ask users
_What would be the perfect mod multipliers_
but
_With what accuracy should scores with mod X tie NM SS scores_,
because lazer score formula is not as intuitive as the stable one. If a mod multiplier is 1.12x then on stable you can estimate that with this mod you need FC with x/1.12 accuracy to match the NM FC with x accuracy. With lazer score this thinking pattern is just wrong and _the best_* mod multipliers will always look too big for most of the users, who still think about them in a _stable_ way.

*the best = maximizing the user's experience that scores on the leaderboards are ordered correctly

**Comment by Purplegaze** [comment]
I feel like interactions between common mod combinations should definitely be accounted for too when surveying a question like "what accuracy should be required to snipe an SS".

In stable, ~94.6% HDDT is needed to snipe an HDHR SS. Do people really agree with raising this to ~96.6% in lazer? (as well as the comparatively high value of 99.3% for DT vs HDHR) What about other contentious mod combinations, such as FL being worth less than HDHR in this proposal?

If further surveying is done, this is definitely something to consider.

**Comment by Livium129** [comment]
Really like this change overall, but had a little bit of an issue with the HDHR-FL weighting. Generally, FL and DTFL scores are much rarer than HDHR and 3mod scores, and giving a larger boost to HDHR disincentivizes playing an already rare mod combo for leaderboards outside of Easy and Normal difficulties. This problem exists on scorev1 as well, and it's also annoying there. So, I reweighted HD and HR to be slightly below FL when combined, giving FL players an ~0.5% boost over HDHR players. My new acc chart, and new multipliers, attached below.

![image](https://github.com/ppy/osu/assets/59489574/6cbd5481-9f7b-4f6d-a064-059f0cb4d849)

This change also had the side effect of giving HDHRFL a 1% disadvantage compared to DTFL and 3mod. HDHRFL's equal weighting to 3mod and DTFL has been a constant annoyance for both DT playerbases, since the lack of rate increase makes it much easier than the other two mod combos in basically all circumstances.

![image](https://github.com/ppy/osu/assets/59489574/83e63a54-fb5d-47c7-a6b5-95510e31ffba)
These are the new multipliers that I used for HD and HR. The HR boost is exactly twice HD's, which is just an arbitrary choice on my part. Feel free to change around if necessary.

**Comment by OwenCMYK** [comment]
I think it would be best to instead gather this data based on real-world data of how many players pass/FC a beatmap depending on different mods. And even then, I personally think mods should be slighty underscored on average compared to their difficulty. Because the difficulty that each mod adds depends heavily on the way the map is designed and the skillset a player has. And I think it's better for mods to be "not worth the effort" than it is for unmodded plays to be "not worth the effort". Because not matter what you make the multiplier, as long as it's above 1, HR will always be worth it to somebody. Even more so with HD. So I feel it's better to ere on the side of caution.

**Comment by ashandelle** [comment]
I know that this discussion hasn't been used in a while but it seems important and I thought it would be fun to set up a new poll since 68 is too few people. I made a new google forms with new mods from Lazer, most of them are optional though.

Link to form: https://docs.google.com/forms/d/e/1FAIpQLSfrDzo-cdaZDXfNRYY4oEfXiA5N3s-m7Oam_lybzzAF677cIA/viewform?usp=sf_link
Link to sheet: https://docs.google.com/spreadsheets/d/1UOKY_L6E7FtY7tqqD6UWqX3jO-fUF19OANe9qmgKC5E/edit?usp=sharing

I'm not very good at making sheets so if you want editing access send a request.

**Reply by MaklovitzLazer** [reply]
Very nice survey, but I found problem not being able to type "100" entry (indicating a mod should have 1.00 multiplier).

Also, idk if DT/HT 0.6x, 0.7x, ..., 2.0x data form user input will be useful; instead I'd rather use user's DT 1.5x and HT 0.75x data and interpolate all other results, so for example if the mean for DT 1.5x will be 92.5% => 1.25 mod multiplier then 1.1x = 1.05, 1.2x = 1.10, ..., 2.0x = 1.50

**Reply by ashandelle** [reply]
It should be possible to enter 100 now. As for the extra data points we can always ignore them if they don't line up nicely but personally I think that 3 points is too few, I made them optional too so I don't think it will be a huge problem.

**Reply by WitherFlower** [reply]
Thanks for sharing such a thorough form ! Will try to share it with as many people as possible.

It would be nice to make the histograms have more columns. I can help with that if needed, you can contact me on discord using the same username as here.

**Comment by ashandelle** [comment]
For difficulty reducing mods is the multiplier just the inverse (a+a^5)/2?

**Reply by MaklovitzLazer** [reply]
For difficulty reducing mods you calculate mod multiplayer the same way as for difficulty increasing and then you take 1/x of it, e.g. 1.25 -> 0.8

**Comment by Purplegaze** [comment]
@ashandelle To clarify since I think my feedback response was misinterpreted:
I think an important thing to also ask people about is when mods snipe _other mods_, not necessarily NM. This may result in different multipliers compared to the ones surveyed with NM comparisons, so it would be interesting to see those differences.

For example, historically it's much more common for a map to have an HDHR SS (or high acc score) as number 1 before it gets its first HDDT FC, so the question of "at what acc does a HDDT FC snipe an HDHR SS" (stable: ~94.9%) (or, perhaps more interestingly, DT only vs. HDHR) is a lot more relevant than DT sniping NM

**Reply by ashandelle** [reply]
Ok

**Reply by ashandelle** [reply]
I'm not sure how to calculate the current accuracy required

**Reply by ashandelle** [reply]
Its added to the form but I'm not making the sheet for it right now

**Comment by ashandelle** [comment]
I've made it so that everyone is an editor but I've restricted it so that the only thing that can be edited is the mods that are listed on the leaderboard.

**Comment by Givikap120** [comment]
DT multiplier should be 1.3x, this is very close to the value derived from stable
And I doubt people think that this values should be lowered
Have you provided surveyed people information about how values are like in scorev1?

**Reply by ashandelle** [reply]
Do you know where I can find the score algorithm for stable?

**Reply by Givikap120** [reply]
> Do you know where I can find the score algorithm for stable?

stable is just "SS value * accuracy", as we don't take combo into account
stable DT multiplier is 1.12x, so nomod SS equals 1/1.12 = 89% DT FC

**Reply by ashandelle** [reply]
If the multiplier and the accuracy for Stable are equivalent then I see no reason to show both, the entire reason for this form is because they aren't on Lazer.

**Reply by ashandelle** [reply]
I think that this is the stable scoring system [ScoreV1](https://osu.ppy.sh/wiki/en/Gameplay/Score/ScoreV1/osu%21) I've taken a look at it and it seems that the exact percentage varies based on the length of the song, longer songs require lower % modded fcs. Although I didn't account for the difficulty multiplier.

**Reply by Givikap120** [reply]
> If the multiplier and the accuracy for Stable are equivalent then I see no reason to show both, the entire reason for this form is because they aren't on Lazer.

Did people knew that 89% DT FC in stable is same as NM SS when voting for 92% value in survey?
If they didn't - I would consider this survey as flawed because participants didn't had a reference point.
This is much better than flat "multiplier" voting as was for CL, but still doesn't represent actual opinion.

**Reply by Givikap120** [reply]
> I think that this is the stable scoring system [ScoreV1](https://osu.ppy.sh/wiki/en/Gameplay/Score/ScoreV1/osu%21) I've taken a look at it and it seems that the exact percentage varies based on the length of the song, longer songs require lower % modded fcs. Although I didn't account for the difficulty multiplier.

Source? Because according to this -> https://osu.ppy.sh/wiki/en/Gameplay/Score/ScoreV1/osu%21 length is not factored in
"Hit value", what represents accuracy, is multiplied flat.
The things that can skew this are:
- amount of sliders (more sliders = less acc influence, because of sliderpoints and reverses giving score)
- amount of spinners (same as sliders but much more drastic on short maps, maybe this is what you meant)
- positions of 100s/50s (the latter they occur on the map - the more they will drop your total score, also known as "scorev1-ning")

**Reply by ashandelle** [reply]
Not length, hit objects which is proportional

**Reply by ashandelle** [reply]
The nonlinearity is added by the combo multiplier

**Reply by Givikap120** [reply]
> The nonlinearity is added by the combo multiplier

combo multiplier is multiplied by acc too
look into formula again

oh, you mean that mod multiplier is not exact, not the acc
I'm sorry then

**Reply by ashandelle** [reply]
The problem is that the weight of a hit object scales with its distance into the map, objects at the end effect accuracy more, and so longer maps have different required mod accuracies.

**Reply by Givikap120** [reply]
> The problem is that the weight of a hit object scales with its distance into the map, objects at the end effect accuracy more, and so longer maps have different required mod accuracies.

Okay, the lowest I've been able to get is 1.105 multiplier (on normal diff of 20 seconds map), what is 90.5% accuracy, still entire 1.5% lower than poll suggests
Real multiplier on the actual map will be like 1.118
